### ISSUE TITLE
Issues with natural python api

### ISSUE THREAD

**Issue by j2kun** [google/or-tools#452]
1. `VariableExpr` has no `__str__` cf. https://github.com/google/or-tools/blob/40ce0077f5a4c1ba3e742dc5c2987a3a8ed63f0d/ortools/linear_solver/linear_solver_natural_api.py#L142

Causes, eg.

```
ipdb> str(arg._SumArray__array[1])
'(79.2 * <ortools.linear_solver.linear_solver_natural_api.VariableExpr object at 0x1060530b8>)'
ipdb> cst = arg._SumArray__array[1]
ipdb> cst._ProductCst__coef
79.2
ipdb> cst._ProductCst__expr
<ortools.linear_solver.linear_solver_natural_api.VariableExpr object at 0x1060530b8>
ipdb> cst._ProductCst__expr._VariableExpr__var
SNAIL,RAW
```

2. When I do a `sum` of an array to construct a constraint, the result is a nested list of lists. This is a problem if the thing being summed is larger than length 300, because then [this function](https://github.com/google/or-tools/blob/40ce0077f5a4c1ba3e742dc5c2987a3a8ed63f0d/ortools/linear_solver/linear_solver_natural_api.py#L195) used to convert the `SumArray` object to a constraint exceeds python's maximum recursion depth. Hence, this should either use a stack/queue to avoid recursion and flatten the list first, or else have the construction of a `SumArray` be a proper list instead of a nested list of lists.

**Comment by j2kun** [comment]
Current workaround for the second issue is to manually import `SumArray` and invoke it directly

```
from ortools.linear_solver.linear_solver_natural_api import SumArray
...
        return SumArray(list_of_variables_with_coefficients)
```

**Comment by lperron** [comment]
Or simply use solver.Sum( ) instead of sum()

Laurent Perron | Operations Research | lperron@google.com | (33) 1 42 68 53
00


2017-08-20 21:06 GMT-07:00 Jeremy Kun <notifications@github.com>:

**Comment by lperron** [comment]
Hello, 

This has been fixed in the latest release (6.5) out today.

Thanks for the report.

**Comment by j2kun** [comment]
Than you! Your work is appreciated :)
